lib/rspec/support/method_signature_verifier.rb

```diff
@@ -26,9 +26,13 @@ def non_kw_args_arity_description
         end
       end
 
-      def valid_non_kw_args?(positional_arg_count)
+      def valid_non_kw_args?(positional_arg_count, optional_max_arg_count=nil)
+        return true if positional_arg_count.nil?
+
+        optional_max_arg_count ||= positional_arg_count
```

Comment:
Can you either collapse this below or use the suggestion for default arg rather than shadowing variables.